http4s/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/Config.scala

```diff
@@ -0,0 +1,293 @@
+package com.snowplowanalytics.snowplow.collectors.scalastream
+
+import java.nio.file.{Files, Path}
+
+import scala.concurrent.duration.FiniteDuration
+import scala.collection.JavaConverters._
+
+import com.typesafe.config.ConfigFactory
+
+import io.circe.config.syntax._
+import io.circe.generic.semiauto._
+import io.circe.Decoder
+import io.circe._
+
+import cats.implicits._
+
+import cats.effect.Sync
+
+import com.monovore.decline.{Command, Opts}
+
+object Config {
+
+  def parse[F[_]: Sync](
+    args: List[String]
+  ): F[Config] = {
+    val appConfig = Opts.option[Path]("config", "Path to HOCON configuration", "c", "config.hocon")
+    val parser    = Command[Path]("collector", "Snowplow application that collects tracking events")(appConfig)
+    for {
+      path <- parser.parse(args) match {
+        case Left(help) =>
+          Sync[F].raiseError(new IllegalArgumentException(s"can't read CLI arguments. $help"))
+        case Right(path) =>
+          Sync[F].pure(path)
+      }
+      config <- parseConfigFile(path)
+    } yield config
+  }
+
+  private def parseConfigFile[F[_]: Sync](path: Path): F[Config] =
+    Either.catchNonFatal(Files.readAllLines(path).asScala.mkString("\n")) match {
+      case Right(raw) =>
+        val config = ConfigFactory
+          .parseString(raw)
+          .resolve()
+          .withFallback(ConfigFactory.load("application.conf"))
+          .withFallback(ConfigFactory.load("reference.conf"))
```

Comment:
For all our other snowplow applications, we accept config parameters with the following order of precedence:

1. System properties set with the `-D` command line option
2. The provided config file (which is optional)
3. `application.conf`
4. `reference.conf`

Here you have allowed for 2-4 but you haven't allowed for system properties.

The way to do this is to sandwich the layers of configuration like the old collector does [here](https://github.com/snowplow/stream-collector/blob/2.9.1/core/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/Collector.scala#L82).

Reply:
I didn't think of 1, I'll update :+1: